File: Chain of Responsibility/Golang/System.go
package main

import (
	"fmt"
	"math/rand"
)

type System struct{}

func (system *System) setNextHandler(handler Handler) {
}

func (system *System) execute(request *Request) *Response {
	fmt.Println("In System")

	return &Response{
		Code:    200,
		Message: "Ok",
		Data:    rand.Intn(100),
	}
}

File: Chain of Responsibility/Dart/System.dart
import 'Handler.dart';
import 'Request.dart';
import 'Response.dart';

import 'dart:math';

class System extends Handler {
  @override
  Response? execute(Request? request) {
    print("In System");

    return Response(200, 'Ok', Random().nextInt(100));
  }
}
